src/medley/core.cljc

```diff
@@ -446,34 +454,60 @@
   ([pred coll]
    (rest (drop-while (complement pred) coll))))
 
+(defn- add-item [coll item]
+  (#?(:clj  .add 
+      :cljs .add
+      :cljr .Add) coll item))
+
+(defn- clear-list [coll]
+  (#?(:clj  .clear
+      :cljr .Clear
+      :cljs .clear) coll))
+
+(defn- is-empty? [coll]
+  #?(:clj  (.isEmpty coll)
+     :cljr (zero? (.-Count coll))
+     :cljs (.isEmpty coll)))
+
+(defn- to-array [coll]
+  (#?(:clj  .toArray
+      :cljr .ToArray
+      :cljs .toArray) coll))
+
+(defn- array-list []
+  #?(:clj  (java.util.ArrayList.)
+     :cljr (System.Collections.ArrayList.)
+     :cljs (cljs.core/array-list)))
```

Comment:
Unfortunately these functions will need to be inlined, otherwise we add in reflection. In other words, prefer:

```clojure
(#?(:clj .clear :cljs .clear :cljr .Clear) part)
```

Or possibly the slightly more concise:

```clojure
(#?(:default .clear :cljr .Clear) part)
```

Over:

```clojure
(clear-list part)
```